### PR TITLE
Example of making batch norm tests compile with CoreOpExecutor

### PR DESCRIPTION
**Pull request by cjolivier01**
Here is some starter diffs.
Not that I don't know how the inputs and outputs are arranged for your API so they're juist arbitrarily chosen in resetForward() and resetBackward().
Also, CoreOpProp (BatchNormCoreOpProp) or its derrivative should somehow be able to select which batch norm operator to run ()gpu, cpu, mkl, cudnn, etc).

Hope this helps,

-Chris

### CODE REVIEW

File: tests/cpp/operator/batchnorm_test.cc [hunk 1]
```diff
@@ -109,67 +106,67 @@ class BNOperatorExecutor : public test::op::LegacyOperatorExecutor<DType, AccRea
 
     // Init the moving data (all mean = 0, all var = 1)
     MSHADOW_TYPE_SWITCH(
-      this->c_.blob_aux_states_[mxnet::op::batchnorm::kMovingMean].type_flag_,
+      this->input_blobs()[mxnet::op::batchnorm::kMovingMean].type_flag_,
```
Comment (cjolivier01) [inline tests/cpp/operator/batchnorm_test.cc:109]: I don;t know what vectors you use for what in your API, so these would need to be matched up correctly.
Probably implementing getBlob(enum data type such as fwd input grad) in the batch norm executor would probably be the easiest way

File: tests/cpp/operator/batchnorm_test.cc [hunk 2]
```diff
@@ -673,15 +671,24 @@ testForwardAndBackward(const bool isGPU,
...
+  const mxnet::op::BatchNormParam& getParam() const {
+    CHECK(false); // Not implemented
+    static mxnet::op::BatchNormParam dummy;
+    return dummy;
+  }
+};
+
 template<typename OperatorExecutor>
-static test::op::OpInfoPair<mxnet::op::BatchNormV1Prop, mxnet::op::BatchNormProp, OperatorExecutor>
+static test::op::OpInfoPair<BatchNormCoreOpProp, BatchNormCoreOpProp, OperatorExecutor>
```
Comment (cjolivier01) [inline tests/cpp/operator/batchnorm_test.cc:684]: This is where the version of the batch norm operator used to be specified, by the OperatorProp derivative.  However it's done in your API would tend to be specified BatchNormCoreOp or its derivative.

File: tests/cpp/operator/batchnorm_test.cc [hunk 3]
```diff
@@ -698,11 +705,14 @@ TEST(BATCH_NORM, Test2DForwardV1V2) {
     DType,
```
Comment (cjolivier01) [inline tests/cpp/operator/batchnorm_test.cc:705]: By the time this test runs as intended, the rest should be mostly trivial.
